### ISSUE TITLE
[QNN 2.25] Significant Accuracy Drop/Bounding Box Overlap in YOLO26n after 8-bit Quantization

### ISSUE THREAD

**Issue by mhzhou2021** [ultralytics/ultralytics#23537]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

After converting and quantizing a YOLO26n model using QNN 2.25, the inference results show severe bounding box overlapping and a significant drop in detection accuracy compared to the original model. 

### Environment

QNN SDK Version: 2.25.0

Host OS: Ubuntu 22.04

Model: YOLO26n 

ONNX Opset: 12

### Minimal Reproducible Example

1. Export ONNX:
```
model.export(format="onnx", imgsz= (192, 256), simplify=True, opset=12)
```
2. Quantize via QNN:
```
MODEL_NAME="yolo26n"

[raw.zip](https://github.com/user-attachments/files/25042221/raw.zip)

qnn-onnx-converter  \
    --input_network models/${MODEL_NAME}.onnx    \
    --output_path models/${MODEL_NAME}_q.cpp \
    --input_list raw/raw_list.txt  \

qnn-model-lib-generator \
    -c models/${MODEL_NAME}_q.cpp \
    -b models/${MODEL_NAME}_q.bin \
    -o libs                \
    -t x86_64-linux-clang
```
3. Run Inference:
```
qnn-net-run --backend qnn_v2.25.0/qaisw-v2.25.0.240728104910_97711-auto-qnx/lib/x86_64-linux-clang/libQnnCpu.so  --model libs/x86_64-linux-clang/libyolo26n_q.so --input_list raw/raw_list.txt
```
4. show detect score >=0.25 results:
<img width="256" height="192" alt="Image" src="https://github.com/user-attachments/assets/52948b80-243e-4d5f-b836-2f922cc61e24" />

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @mhzhou2021, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

This is an automated response 🤖—an Ultralytics engineer will also assist soon.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. From your report, you’ve shared the high-level steps, but an MRE would ideally include a small, self-contained sample (a few images + labels or a tiny public subset), the exact export command/script used end-to-end, and all relevant logs so we can reproduce the QNN quantization accuracy drop and box overlap behavior on our side.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
The output looks like wrong postprocessing. The confidence is greater than 1.

**Comment by mhzhou2021** [comment]
@Y-T-G  Thank you！ Yes, it is likely a quantization issue. However, the YOLO26n-pose model uses the same quantization method, and its test results are correct.

<img width="256" height="192" alt="Image" src="https://github.com/user-attachments/assets/8e3f1449-699e-4fa8-a752-5b5026973b56" />

**Comment by Y-T-G** [comment]
You can try exporting with `agnostic_nms=True`

**Comment by mhzhou2021** [comment]
> You can try exporting with `agnostic_nms=True`
@Y-T-G I tested it and the same issue still exists. However, inference with the exported ONNX model is correct.

<img width="256" height="192" alt="Image" src="https://github.com/user-attachments/assets/b28d5acb-ad08-412a-a349-b0d84fd4886a" />

**Comment by Y-T-G** [comment]
Did you export with the latest Ultralytics?

**Comment by mhzhou2021** [comment]
> Did you export with the latest Ultralytics?
@Y-T-G  ultralytics 8.4.9

**Comment by Y-T-G** [comment]
You will probably have to remove `postprocess`

https://github.com/ultralytics/ultralytics/blob/1da4e933181f0dfe983349c2b1dcda81dba094f8/ultralytics/nn/modules/head.py#L160

and perform postprocess manually in your code

**Comment by mhzhou2021** [comment]
> You will probably have to remove `postprocess`
> 
> [ultralytics/ultralytics/nn/modules/head.py](https://github.com/ultralytics/ultralytics/blob/1da4e933181f0dfe983349c2b1dcda81dba094f8/ultralytics/nn/modules/head.py#L160)
> 
> Line 160 in [1da4e93](/ultralytics/ultralytics/commit/1da4e933181f0dfe983349c2b1dcda81dba094f8)
> 
>  y = self.postprocess(y.permute(0, 2, 1)) 
> and perform postprocess manually in your code
@Y-T-G  Yes, I modified the post-processing logic during the ONNX export phase.
```
def detect_postprocess(self, preds: torch.Tensor) -> torch.Tensor:
    return preds
def export_detect_model(model_path, imgsz=INPUT_SIZE, opset=12):
    # load the model
    model = YOLO(model_path)
    for layer in model.modules():
        if "ultralytics.nn.modules.head.Detect" in str(type(layer)) and hasattr(layer, "postprocess"):
            layer.postprocess = types.MethodType(detect_postprocess, layer)
    model.export(format="onnx", imgsz=imgsz, simplify=True, opset=12)
```

**Comment by Y-T-G** [comment]
So does it work after modification?

**Comment by mhzhou2021** [comment]
> So does it work after modification?
@Y-T-G The results is correct! thank you!

<img width="256" height="192" alt="Image" src="https://github.com/user-attachments/assets/10e1a675-a48c-4ab8-be65-dff04848f4bd" />

**Comment by Y-T-G** [comment]
Great. Can you write the export step by step so other users can use it later?

**Comment by mhzhou2021** [comment]
> Great. Can you write the export step by step so other users can use it later?

OK!
```
import types
import torch
from ultralytics import YOLO
def detect_postprocess(self, preds: torch.Tensor) -> torch.Tensor:
    # boxes, scores = preds.split([4, self.nc], dim=-1)
    # scores, conf, idx = self.get_topk_index(scores, self.max_det)
    # boxes = boxes.gather(dim=1, index=idx.repeat(1, 1, 4))
    # return torch.cat([boxes, scores, conf], dim=-1)
    return preds.permute(0, 2, 1)
if __name__ == '__main__':
    model_path = "./models/yolo26n.pt"
    # load the model
    model = YOLO(model_path)
    # replace ultralytics.nn.modules.head.Detect.postprocess with detect_postprocess
    for layer in model.modules():
        if "ultralytics.nn.modules.head.Detect" in str(type(layer)) and hasattr(layer, "postprocess"):
            layer.postprocess = types.MethodType(detect_postprocess, layer)
    
    model.export(format="onnx", imgsz=imgsz, simplify=True, opset=12)
```

**Comment by glenn-jocher** [comment]
Thanks for writing this up—one small clarification for anyone copying it: `Detect.forward()` already calls `self.postprocess(y.permute(0, 2, 1))`, so if your goal is to export *raw* head outputs in shape `(B, num_anchors, 4 + nc)` for QNN-side decoding/NMS, your patched `postprocess()` typically should just `return preds` (no extra `permute`), and then you can mirror Ultralytics’ decoding/top-k/NMS logic externally as needed (see the reference implementations of [`Detect.postprocess()` and `Detect.get_topk_index()`](https://docs.ultralytics.com/reference/nn/modules/head/)).
